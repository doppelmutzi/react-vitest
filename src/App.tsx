import { useEffect, useState } from 'react'
import reactLogo from './assets/react.svg'
import vitestLogo from '/vitest.svg'
import './App.css'
import { fetchQuote, Quote } from './quote.service'

function App() {
  const [count, setCount] = useState(0)


  const [quote, setQuote] = useState<Quote>();
  useEffect(() => {
    const getQuote = async () => {
      const quote = await fetchQuote();
      setQuote(quote);
    };
    getQuote();
  }, []);
  

  return (
    <>
      <div>
        <a href="https://vitest.dev/" target="_blank">
          <img src={vitestLogo} className="logo" alt="Vitest logo" />
        </a>
        <a href="https://react.dev" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>Vitest + React ü§ù</h1>
      <section>
        <button onClick={() => setCount((count) => count + 1)}>
          count is {count}
        </button>
        <h2>Quote of the Day</h2>
        <p>{quote ? quote.quote : 'Loading...'}</p>
      </section>
    </>
  )
}

export default App
