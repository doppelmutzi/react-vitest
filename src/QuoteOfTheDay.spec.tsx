import { expect, test, vi } from "vitest";
import App from "../src/App";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
// import * as quoteService from "./quote.service";

vi.mock("./quote.service", () => ({
  fetchQuote: vi.fn().mockResolvedValue({
    id: 1,
    quote: "This is a test quote",
    author: "Test Author",
  }),
}));

test("renders quote after fetching", async () => {
  // const mockQuote = {
  //     id: 1,
  //     quote: "This is a test quote",
  //     author: "Test Author",
  //   };
  //   vi.mocked(quoteService.fetchQuote).mockResolvedValue(mockQuote);

  const { container, getByText } = render(<App />);
  expect(container).toMatchSnapshot();

  // Wait for the quote to be fetched and rendered
  await waitFor(() => expect(getByText("This is a test quote")).toBeDefined());

  const button = screen.getByText("count is 0");
  fireEvent.click(button);
  expect(container).toMatchSnapshot();
});
