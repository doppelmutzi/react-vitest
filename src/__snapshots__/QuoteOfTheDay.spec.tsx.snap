// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders quote after fetching 1`] = `
<div>
  <div>
    <a
      href="https://vitest.dev/"
      target="_blank"
    >
      <img
        alt="Vitest logo"
        class="logo"
        src="/vitest.svg"
      />
    </a>
    <a
      href="https://react.dev"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="/src/assets/react.svg"
      />
    </a>
  </div>
  <h1>
    Vitest + React 🤝
  </h1>
  <section>
    <button>
      count is 
      0
    </button>
    <h2>
      Quote of the Day
    </h2>
    <p
      data-testid="quote"
    >
      Loading...
    </p>
  </section>
</div>
`;

exports[`renders quote after fetching 2`] = `
<div>
  <div>
    <a
      href="https://vitest.dev/"
      target="_blank"
    >
      <img
        alt="Vitest logo"
        class="logo"
        src="/vitest.svg"
      />
    </a>
    <a
      href="https://react.dev"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="/src/assets/react.svg"
      />
    </a>
  </div>
  <h1>
    Vitest + React 🤝
  </h1>
  <section>
    <button>
      count is 
      1
    </button>
    <h2>
      Quote of the Day
    </h2>
    <p
      data-testid="quote"
    >
      This is a test quote
    </p>
  </section>
</div>
`;
